drivers/regulator/regulator_shell.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/regulator.h>
+#if CONFIG_REGULATOR_PMIC
+#include <zephyr/drivers/regulator/consumer.h>
+#endif
+
+LOG_MODULE_REGISTER(regulator_shell, CONFIG_REGULATOR_LOG_LEVEL);
+
+static int cmd_reg_en(const struct shell *sh, size_t argc, char **argv)
+{
+	struct onoff_client cli;
+	const struct device *reg_dev;
+	int ret;
+
+	if (argc != 2) {
+		shell_print(sh, "usage: %s [dev name]", argv[0]);
+		return -EINVAL;
+	}
+	reg_dev = device_get_binding(argv[1]);
+	if (reg_dev == NULL) {
```

Comment:
You need to use `device_is_ready()` instead of checking against `NULL` all over.